website/client/src/libs/highlightUsers.js

```diff
@@ -1,7 +1,7 @@
 import escapeRegExp from 'lodash/escapeRegExp';
 
 const optionalAnchorTagRegExStr = '(<\\w[^>]*)?'; // everything including the anchor tag is recognized
-const mentionRegExStr = '(@[\\w-]+)';
+const mentionRegExStr = '(@[<em>\\w-</em>]+)';
```

Comment:
Between square brackets is a character class, i.e. "Match any of the characters between these brackets". This way it's actually not a legal regular expression. 

I would recommend always trying out your regular expression strings in a place like https://regex101.com/

Adding the characters in the character class (like this `(@[<>\\/\\w-]+)` would also match things like `@<<<<<<<<<` or `@<a` which is not desirable. I think to get the effect you're looking for you can try something like this `(@(?:<(?:em|strong)>)?[\\w-]+(?:</(?:em|strong)>)?`. It maintains the original semantics of the regex but enhances it with optional _em_ or _strong_ tag matches. All the `?:` characters are to prevent these new bracket groups from messing up the existing grouping semantics.

Reply:
Gave the expanded regular expression a whirl on regex101: https://regex101.com/r/4nFDHZ/1

It seems to work well (BTW, in the comment above I wrote it as it should be injected in the string. The syntax on regex101 is what it would look like as RegEx literal. The difference is in the escaping of backslashes (\\) ).

Reply:
I'm not very used to RegExs so thank you for your suggestion! I actually did use that website to check my RegEx but I think your RegEx is much more to the effect that I was trying to get at. I'll add it in and check that everything still works!